en/drafts/originals/agent-based-model-communication-networks.md

```diff
@@ -629,26 +633,26 @@ class LetterModel(mesa.Model):
 ```
 
 
-After every step of the model, the datacollector will collect and store each agent's letters_sent and letters_received value.
+After every step of the model, the data collector will collect and store each agent's `letters_sent` and `letters_received` values.
 
-We run the model just as we did above. The `DataCollector` can export the data it has collected as a pandas `DataFrame`, for easy interactive analysis.
+We run the model just as we did above:
 
 ```python
 model = LetterModel(50, 10, 10)
 for i in range(100):
     model.step()
 ```
 
-We can now get the agent-letters data like this:
+The `DataCollector` can export the data it has collected as a pandas `DataFrame`, allowing for easy interactive analysis. We can get the `agent_letters` data like this:
```

Comment:
Could we add a few words to explain/confirm what the `agent_letters` data refers to?

Reply:
```suggestion
The `DataCollector` can export the data it has collected as a pandas `DataFrame`, allowing for easy interactive analysis. We can get the `agent_letters` data, which currently records information on letters received and letters sent as well as the total number of letters, like this:
```

Reply:
would this suffice? or would you prefer a separate sentence for readability?

Reply:
I'm still slightly unsure. Does the `agent_letters` data contain the number of letters sent and received by each agent at each step, as well as the combination of these numbers i.e. the total number of letters sent and the total number of letters received by each agent at the end of the simulation?

This total seems to be what Figure 5 shows below: the "Number of letters sent and received by a selection of agents, at step 100 of the simulation".

Reply:
Yes, almost correct! On the agent level (ie, per agent) letters sent and received are collected; on the model level (ie, for everything) the total number of "letters received" is collected. For the latter, you can look for the compute_received_letters function.

self.datacollector = mesa.DataCollector(
            agent_reporters={ -> per agent
                "Letters_sent": "letters_sent", 
                "Letters_received": "letters_received"
            },
            model_reporters={"All letters":compute_received_letters} -> for the whole model / every agent combined
        )

---

```python
def compute_received_letters(model): -> the total / model based amount
    number_of_received_letters = 0
    for agent in model.schedule.agents:
        number_of_received_letters += agent.letters_received
    return number_of_received_letters
```

Reply:
Thanks @merrygin – I've written change it to:

We can get the `agent_letters` data (the combined agent-level and model-level data) like this: [...]

Does that make sense?

Reply:
Yes that's great